packages/volto/src/components/manage/Sidebar/SidebarPopup.jsx

```diff
@@ -16,18 +16,26 @@ const SidebarPopup = (props) => {
     onClose();
   };
 
+  const handleEscapeKey = (e) => {
+    if (open && e.key === 'Escape') {
+      onClose();
+      e.stopPropagation();
+    }
+  };
+
   const [isClient, setIsClient] = React.useState(false);
   React.useEffect(() => {
     setIsClient(true);
   }, []);
 
   React.useEffect(() => {
     document.addEventListener('mousedown', handleClickOutside, false);
+    document.addEventListener('keyup', handleEscapeKey, false);
```

Comment:
Just a comment why adding the `keyup`?

Reply:
@nileshgulia1 ^^

Reply:
@sneridagh I had to cut down the propagation to https://github.com/plone/volto/blob/1b8328cc0029f092525f6791d5458665b5008a4b/packages/volto/src/helpers/Utils/useDetectClickOutside.js#L52 that removes the focus on selected block.